idaes/core/util/math.py

```diff
@@ -191,3 +191,16 @@ def safe_log(a, eps=1e-4):
         approximately log(max(a, eps))
     """
     return log(smooth_max(a, eps, eps=eps))
+
+
+def smooth_heaviside(x, k):
+    """
+    Provides a smooth, continuous approximation of a discontinuous function
```

Comment:
"Provides a smooth, continuous approximation of the Heaviside step function using the logistic function"